### PR TITLE
Making effective learning rate accessible for user in Adam

### PR DESCRIPTION
**Pull request by ahmedhshahin**
Hello,

In one of our projects, we wanted to retrieve the effective learning rate between epochs. Since we were using Adam optimizer, this learning rate is changing dynamically between epochs according to Adam's optimization technique. In the current implementation of PyTorch, only the initial learning rate is accessible for the user. To get the effective learning rate, I added it to the param_groups dictionary in Adam's class. I found many people asking for  a way to access it on stackoverflow and torch forum. So, I made this pull request to include this modification.
Hope it helps.

Regards,

### CODE REVIEW

**Review by vincentqb — CHANGES_REQUESTED** [review]
This change is also fairly easy to do for someone who wants it: copy-paste adam, and return any variables of interests. I would favor that instead.

My main concern is that, if we decide to do this, we need to update every optimizers to support this. We also probably also want to do other variables like `effective_weigth_decay` etc. This would add more code complexity. I would hold off on this PR for now.

PS. This does touch on the idea of having optimizer building blocks that can be mixed and matched instead of providing monolithic implementations of optimizers.
